cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -56,6 +58,13 @@ class ARROW_EXPORT ScalarAggregateOptions : public FunctionOptions {
   uint32_t min_count;
 };
 
+class ARROW_EXPORT CountAllOptions : public FunctionOptions {
```

Comment:
Hmm, what is this for? If a given function doesn't support any specific options, then a `nullptr` FunctionOptions should simply be passed.

Reply:
Unfortunately, I can't pass `nullptr` as the "default options" in the kernel configuration. I tried, but then it causes crashes because when you pass no options in the kernel calls, `options` is set to the default options as after some point, `options` is expected to be non-null.

Reply:
Hmm, hopefully this can be fixed, then. Some other compute functions don't have an associated options class.

Reply:
I will open a separate PR fixing this then.

cc @westonpace @vibhatha

Reply:
It's a small fix, so I'm pushing a commit here.